### DISCUSSION TITLE
I Lost Access to Copilot Preview

### DISCUSSION THREAD

**Discussion by rtadijar** [community/community#19052] [Copilot Conversations] [answered]
I had gotten access to Github Copilot on 26/10/2021 and have been using it since.

Starting today, VSCode tells me I am not authorized and Github is offering me to enter the waitlist. 

The VSCode extension gives the following output log:
```
[INFO] [auth] [2022-06-21T16:32:15.935Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-06-21T16:32:15.936Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"
```

My extensions are updated, and since Github is also telling me about entering the waitlist, I assume this is not a version issue.

**Comment by rauzerfabiano** [comment]
It might have something to do with the changes they are making. They want to put the copilot as paid after 2 months of free trial.

https://github.com/github-copilot/tp_signup

**Reply by rtadijar** [reply]
I see, I assume this is the case. Some heads-up or confirmation would be nice.

**Reply by rauzerfabiano** [reply]
Yes, I totally agree with you.

**Comment by revenkroz** [comment, marked as answer]
Just visit https://github.com/github-copilot/tp_signup and update data. It should be fixed right after that.

**Reply by rtadijar** [reply]
That link is broken for me.

**Reply by revenkroz** [reply]
Sorry. Updated.

**Reply by rtadijar** [reply]
The link offers me to opt in for the paid version. I do not see the 'update data' option.

**Reply by revenkroz** [reply]
Click on "Continue without payment details." to use free until August 22

**Reply by rtadijar** [reply]
Got it back. Thank you.

**Reply by hiyoi** [reply]
I click Sign up for Github Copilot in extension pop panel.  the Browser will go to https://github.com/github-copilot/signup?editor=copilot_1.30.6165 , finally redirect to https://github.com/settings/billing.
but i still got 
`[INFO] [auth] [2022-06-23T10:37:53.383Z] Invalid copilot token: missing token: 403 `
`[ERROR] [default] [2022-06-23T10:37:53.385Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"`

i already visit https://github.com/github-copilot/tp_signup and update data.

**Reply by dhust** [reply]
It's a paid service now.

**Reply by fazofazaal** [reply]
> I click Sign up for Github Copilot in extension pop panel. the Browser will go to https://github.com/github-copilot/signup?editor=copilot_1.30.6165 , finally redirect to https://github.com/settings/billing. but i still got `[INFO] [auth] [2022-06-23T10:37:53.383Z] Invalid copilot token: missing token: 403 ` `[ERROR] [default] [2022-06-23T10:37:53.385Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"`
> 
> i already visit https://github.com/github-copilot/tp_signup and update data.

I initially tried to pay but due to a limitation with my bank on foreign purchases, I had reached my payment limit and the transaction got declined by my bank. So then I tried to continue without payment but tried and failed multiple times to connect. I'm getting the same errors.

[INFO] [auth] [2022-06-25T16:13:16.359Z] Invalid copilot token: missing token: 403 
[ERROR] [default] [2022-06-25T16:13:16.360Z] GitHub Copilot could not connect to server. Extension activation failed: "User not authorized"

**Reply by davecheney** [reply]
@hiyoi 
<img width="475" alt="Screen Shot 2021-11-30 at 8 03 57 am" src="https://user-images.githubusercontent.com/7171/143942796-0c2b6d34-e0f4-46fe-b349-f26087b2c5ed.png">
Please try signing out, copilot will prompt you to sign back in.

**Reply by hiyoi** [reply]
@davecheney i try visit https://github.com/github-copilot/tp_signup and got 
<img width="735" alt="image" src="https://user-images.githubusercontent.com/2487038/176076467-97aa5211-5d61-48df-b018-3051d89377bc.png">

sign out account from vscode,  uninstall copilot extension and reinstall.  copilot prompt  me sign in,  but always redirect to https://github.com/settings/billing , still doesn't work.
<img width="1143" alt="image" src="https://user-images.githubusercontent.com/2487038/176077287-75864ad0-bbab-4715-a415-f4d1fea2ef71.png">


my copilot panel show: 
<img width="787" alt="image" src="https://user-images.githubusercontent.com/2487038/176076933-c2ed2f00-8416-4975-bfcd-a5b501db1e00.png">

**Reply by davecheney** [reply]
@hiyoi your billing status is broken. Please write to support and ask them to reset your billing attempts.

**Comment by dhust** [comment]
I'm sure we've all done that multiple times. Still doesn't work.

**Reply by jhmaster2000** [reply]
Same issue here, I am stuck on a loop of trying to activate it, it says it was successful and will be free until August 22, I go back to VSCode, restart it, and still not authorized. Unmark this is as resolved, it's very much not.

**Reply by rtadijar** [reply]
I believe you need to apply for the free trial until the 22th. Our previous access was revoked in favor of this opt-in.

EDIT: I see now that this doesn't work for you. I'd suggest resetting your Copilot extensions, signing out of your account in VSCode, etc.

At any rate, your problem is slightly different than mine, but you should definitely report it!
